.changelog/unreleased/improvements/169-tx_index-pruning.md

```diff
@@ -0,0 +1,2 @@
+- `[state/indexer]` Add pruning for the tx_index.db. 
```

Comment:
```suggestion
- `[state/indexer]` Add transaction and block index pruning
```

Users don't necessarily know what `tx_index.db` is, and this PR does more than just facilitate transaction index pruning. Changelog entries are not for us, they're for our users.

Reply:
Done